modules/curl/8.8.0.bcr.1/patches/add_build_file.patch

```diff
@@ -0,0 +1,334 @@
+diff --git a/BUILD.bazel b/BUILD.bazel
+new file mode 100644
+index 0000000..ff61c6f
+--- /dev/null
++++ b/BUILD.bazel
+@@ -0,0 +1,328 @@
++# Copyright The OpenTelemetry Authors
++# SPDX-License-Identifier: Apache-2.0
++
++load("@bazel_skylib//rules:common_settings.bzl", "bool_flag", "string_flag")
++load("@bazel_skylib//lib:selects.bzl", "selects")
++
++package(features = ["no_copts_tokenization"])
++
++bool_flag(
++    name = "use_mbedtls",
++    build_setting_default = False,
++)
++
++config_setting(
++    name = "use_mbedtls_setting",
++    flag_values = {":use_mbedtls": "true"},
++)
++
++config_setting(
++    name = "not_use_mbedtls_setting",
++    flag_values = {":use_mbedtls": "false"},
++)
++
++selects.config_setting_group(
++    name = "use_boringssl",
```

Comment:
is boring ssl the upstream default with cmake as well?

Reply:
AFAIK, `curl` with CMake defaults to using `*SSL`:

https://github.com/curl/curl/blob/5ab2eda80360c55db4072f9f4747d10a842b3233/CMakeLists.txt#L408-L412

However, most likely this is OpenSSL, but it could be LibreSSL, or WolfSSL, or something else.  On Windows it defaults to schannel, and on macOS to Secure Transport.